### PR TITLE
Update references to renamed jupyterhub-home-nfs chart

### LINKED ISSUE #4889: Rename jupyter-home-nfs into jupyterhub-home-nfs
In https://github.com/sunu/jupyter-home-nfs/issues/2 we agreed to rename the project from `jupyter-home-nfs` to `jupyterhub-home-nfs`. This task is to make that happen

## Definition of done

- [x] Repo is renamed
- [x] Helm chart is renamed
- [x] 2i2c-org/infrastructure repo is updated to use renamed helm chart
- [x] Any python code and documentation referencing old name is renamed

### PR DESCRIPTION
**Pull request by sgibson91**
- fixes #4889

### CODE REVIEW

**Comment by sgibson91** [comment]
Hey @sunu could you help me understand what's going wrong with the GHCR package here please?

**Comment by sunu** [comment]
@sgibson91 this is failing because the published packages in https://github.com/orgs/2i2c-org/packages?repo_name=jupyterhub-home-nfs are all private. Can we make them public?

**Comment by sgibson91** [comment]
> Can we make them public?

Yes, please do!

**Comment by sgibson91** [comment]
Is there a way to make the workflow in that repo push them publicly by default?

**Comment by sgibson91** [comment]
Ah ok, I'm on it

**Comment by sunu** [comment]
> Is there a way to make the workflow in that repo push them publicly by default?
Once the package visibility is set to public, any new package pushed by the workflow will be public as well. But I don't know how to make the package public by default from the beginning of the top of my head

**Comment by sgibson91** [comment]
I changed the setting at the org level

**Comment by sunu** [comment]
Cool, I changed all the packages to be public.
image

**Comment by sgibson91** [comment]
GHA seems to be struggling to re-run the failed jobs, so let's leave that for a bit.

**Comment by sgibson91** [comment]
> Once the package visibility is set to public, any new package pushed by the workflow will be public as well.

This is fine. It's a one-time action per package, and I don't think we'll be adding more soon :)

**Comment by sgibson91** [comment]
Hmmmm, still can't pull the image :/

File: helm-charts/basehub/Chart.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ dependencies:
     version: "2024.1.0"
     repository: "https://helm.dask.org/"
     condition: dask-gateway.enabled
-  - name: jupyter-home-nfs
+  - name: jupyterhub-home-nfs
     version: 0.0.5
```
Comment (sunu) [inline helm-charts/basehub/Chart.yaml:29]: The `0.0.5` version isn't ported over to the new package under 2i2c Github org. I just pushed `0.0.6` tag to get a new version to use. Let's use that here.
```suggestion
    version: 0.0.6
```

**Comment by sunu** [comment]
@sgibson91 I made a fix by creating a new release of the helm chart and using that here. The previous versions were only accessible under the old name.

**Comment by sgibson91** [comment]
Thanks for the assist @sunu!

**Pull request merged**